content/posts/GSoC_2021_Quarter/index.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "GSoC'21: Quarter Progress"
+date: 2021-08-03T18:48:00+05:30
+draft: false
+categories: ["News", "GSoC"]
+description: "Quarter Progress with Google Summer of Code 2021 project under NumFOCUS: Aitik Gupta"
+displayInList: true
+author: Aitik Gupta
+
+resources:
+- name: featuredImage
+  src: "AitikGupta_GSoC.png"
+  params:
+    showOnTop: true
+---
+
+**“<ins>Matplotlib, I want 多个汉字 in between my text.</ins>”**
+
+Let's say you wanted some (多个汉字 -> multiple Chinese characters) in between your English text, to be rendered by Matplotlib.
+
+Or conversely, let's say you use a Chinese font with Matplotlib, but you had English text in between (which is quite common).
+
+> Assumption: the Chinese font doesn't have those English glyphs, and vice versa
+
+With this short writeup, I'll talk about how does a migration from a font-first to a text-first approach in Matplotlib looks like, which ideally solves the above problem.
+### Have the fonts?
+Logically, the very first step to solving this would be to ask whether you _have_ multiple fonts, right?
+
+Matplotlib doesn't ship [CJK](https://en.wikipedia.org/wiki/List_of_CJK_fonts) (Chinese Japanese Korean) fonts, which ideally contains these Chinese glyphs. It does try to cover most grounds with the [default font](https://matplotlib.org/stable/users/dflt_style_changes.html#normal-text) it ships with, however.
+
+So if you don't have a font to render your Chinese characters, go ahead and install one! Matplotlib will find your installed fonts.

```

Comment:
Is this true out of the box or do you need to rebuild the font cache?

Reply:
We do need to rebuild the cache, but the statement was meant to be a generic one.. that Matplotlib _can_ find your custom installed fonts.

Reply:
Modified:
```txt
Matplotlib will find your installed fonts (after rebuilding the cache, that is).
```